### ISSUE TITLE
Segfault on from triro.ray.ray_optix import RayMeshIntersector

### ISSUE THREAD

**Issue by XDynames** [lcp29/trimesh-ray-optix#4]
Just trying to set this up in a project that uses trimesh to do some raycasting and I am getting a segmentation fault when importing your package.

I am using CUDA 12.3, Pytorch 2.3 with Optix 8.0.0 on ubunutu in the nvcr.io/nvidia/pytorch:24.02-py3 docker container

Can you think of anything off the bat that could be wrong or how to chase down that problem the further?
I figure it must be some sort of version conflict

`Segmentation fault: address not mapped to object at address (nil)`

Did some poking around in `gdb` it get thrown by `triro.so` in `optixDeviceContextCreate ()`

**Comment by HS2290** [comment]
Same problem.

**Comment by lcp29** [comment]
Sorry for the late reply. Is there more outputs form the program? And perhaps a minimal code set to reproduce the problem:)
I will try getting the docker run first so maybe let me know what GPUs are you using?

**Comment by HS2290** [comment]
> Sorry for the late reply. Is there more outputs form the program? And perhaps a minimal code set to reproduce the problem:) I will try getting the docker run first so maybe let me know what GPUs are you using?

Thanks for your reply. I am using H100 Server, CUDA 12.1, Ubuntu 22.04, python 3.10. 
My environment only includes python==3.10 and triro==1.3.0 (successfully installed).
When I finish the installation successfully and run 'import triro.ray', the code will only output 'Segmentation fault (core dumped)'. I run 'import triro' is OK, but run 'import triro.ray' fails :(

**Comment by lcp29** [comment]
> > Sorry for the late reply. Is there more outputs form the program? And perhaps a minimal code set to reproduce the problem:) I will try getting the docker run first so maybe let me know what GPUs are you using?
> 
> Thanks for your reply. I am using H100 Server, CUDA 12.1, Ubuntu 22.04, python 3.10. My environment only includes python==3.10 and triro==1.3.0 (successfully installed). When I finish the installation successfully and run 'import triro.ray', the code will only output 'Segmentation fault (core dumped)'. I run 'import triro' is OK, but run 'import triro.ray' fails :(

Than you for the feedback! I will investigate into it.

**Comment by kangyiyang** [comment]
I found that I get a segmentation fault error on the A40, but not on the 4090D. Is this related to the GPU as well? I am using CUDA 11.8, PyTorch 2.1.2, and OptiX 8.0.0

**Comment by XDynames** [comment]
I had the seg fault issue on a 4090
Maybe it’s more to do with the exact combination of drivers/dependancies at compile time?

**Comment by kangyiyang** [comment]
Yes, GPU hardware may be required to match the version of some libraries. But I'm not sure either. Maybe you can change the GPU environment and try more



---Original---
From: "James ***@***.***&gt;
Date: Sun, Jan 5, 2025 07:52 AM
To: ***@***.***&gt;;
Cc: "Yikang ***@***.******@***.***&gt;;
Subject: Re: [lcp29/trimesh-ray-optix] Segfault on from triro.ray.ray_optiximport RayMeshIntersector (Issue #4)




 
I had the seg fault issue on a 4090
 Maybe it’s more to do with the exact combination of drivers/dependancies at compile time?

**Comment by lcp29** [comment]
I got the same problem with OptiX 8.1.0, which requires the driver version >= 555 but mine is 550. After switching to OptiX 8.0.0 (>=535) it worked fine.

**Comment by kangyiyang** [comment]
Great, I'm glad to hear that.



---Original---
From: ***@***.***&gt;
Date: Sun, Jan 5, 2025 15:55 PM
To: ***@***.***&gt;;
Cc: "Yikang ***@***.******@***.***&gt;;
Subject: Re: [lcp29/trimesh-ray-optix] Segfault on from triro.ray.ray_optiximport RayMeshIntersector (Issue #4)




 
I got the same problem with OptiX 8.1.0, which requires the driver version &gt;= 555 but mine is 550. After switching to OptiX 8.0.0 (&gt;=535) it worked fine.
